tests/components/pushbullet/test_notify.py

```diff
@@ -1,301 +1,273 @@
-            "message": "Test Message",
-            "target": ["device/DESKTOP"],
-        }
-        self.hass.services.call(notify.DOMAIN, "test", data)
-        self.hass.block_till_done()
-        assert mock.called
-        assert mock.call_count == 1
+async def test_pushbullet_config_bad(hass):
+    """Test set up the platform with bad/missing configuration."""
+    config = {notify.DOMAIN: {"platform": "pushbullet"}}
+    with assert_setup_component(0) as handle_config:
+        assert await async_setup_component(hass, notify.DOMAIN, config)
+        await hass.async_block_till_done()
+    assert not handle_config[notify.DOMAIN]
 
-        expected_body = {
-            "body": "Test Message",
-            "device_iden": "identity1",
-            "title": "Test Title",
-            "type": "note",
-        }
-        assert mock.last_request.json() == expected_body
 
-    @requests_mock.Mocker()
-    @patch.object(
-        PushBullet,
-        "_get_data",
-        return_value=json.loads(load_fixture("pushbullet_devices.json")),
-    )
-    def test_pushbullet_push_devices(self, mock, mock__get_data):
-        """Test pushbullet push to default target."""
-        config = {
-            notify.DOMAIN: {
-                "name": "test",
-                "platform": "pushbullet",
-                "api_key": "MYFAKEKEY",
-            }
-        }
-        with assert_setup_component(1) as handle_config:
-            assert setup_component(self.hass, notify.DOMAIN, config)
-        assert handle_config[notify.DOMAIN]
-        mock.register_uri(
-            requests_mock.POST,
-            "https://api.pushbullet.com/v2/pushes",
-            status_code=200,
-            json={"mock_response": "Ok"},
-        )
-        data = {
-            "title": "Test Title",
-            "message": "Test Message",
-            "target": ["device/DESKTOP", "device/My iPhone"],
+async def test_pushbullet_push_default(hass, requests_mock, mock_pushbullet):
+    """Test pushbullet push to default target."""
+    config = {
+        notify.DOMAIN: {
+            "name": "test",
+            "platform": "pushbullet",
+            "api_key": "MYFAKEKEY",
         }
-        self.hass.services.call(notify.DOMAIN, "test", data)
-        self.hass.block_till_done()
-        assert mock.called
-        assert mock.call_count == 2
-        assert len(mock.request_history) == 2
+    }
+    with assert_setup_component(1) as handle_config:
+        assert await async_setup_component(hass, notify.DOMAIN, config)
+        await hass.async_block_till_done()
+    assert handle_config[notify.DOMAIN]
+    requests_mock.register_uri(
+        "POST",
+        "https://api.pushbullet.com/v2/pushes",
+        status_code=200,
+        json={"mock_response": "Ok"},
+    )
+    data = {"title": "Test Title", "message": "Test Message"}
+    await hass.services.async_call(notify.DOMAIN, "test", data)
+    await hass.async_block_till_done()
+    assert requests_mock.called
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6285789)